src/parallel_queue.rs

```diff
@@ -0,0 +1,290 @@
+                &vk::CommandPoolCreateInfo::builder()
+                    .queue_family_index(self.1.queue_family_index)
+                    .flags(
+                        vk::CommandPoolCreateFlags::RESET_COMMAND_BUFFER
+                            | vk::CommandPoolCreateFlags::TRANSIENT,
+                    ),
+                None,
+            )
+            .unwrap();
+        let spare_cmds = device
+            .allocate_command_buffers(
+                &vk::CommandBufferAllocateInfo::builder()
+                    .command_pool(cmd_pool)
+                    .command_buffer_count(32),
+            )
+            .unwrap();
+        Handle {
+            send: self.0,
+            shared: self.1,
+            cmd_pool,
+            spare_cmds,
+            in_flight: VecDeque::new(),
+            active: None,
+        }
+    }
+}
+
+pub struct Handle {
+    send: mpsc::Sender<Work>,
+    shared: Arc<Shared>,
+    cmd_pool: vk::CommandPool,
+    spare_cmds: Vec<vk::CommandBuffer>,
+    in_flight: VecDeque<Work>,
+
+    active: Option<Work>,
+}
+
+impl Handle {
+    /// # Safety
+    /// `device` must match that passed to [`ParallelQueue::new`] and no work may be in flight
+    pub unsafe fn destroy(&mut self, device: &Device) {
+        device.destroy_command_pool(self.cmd_pool, None);
+    }
+
+    #[inline]
+    pub fn semaphore(&self) -> vk::Semaphore {
+        self.shared.semaphore
+    }
+
+    #[inline]
+    pub fn cmd_pool(&self) -> vk::CommandPool {
+        self.cmd_pool
+    }
+
+    /// Begin recording commands
+    ///
+    /// Must be called before calls to `cmd()` or `time()`. `end()` must be called soon after to
+    /// avoid stalling the queue.
+    ///
+    /// # Safety
+    /// `device` must match that passed to [`ParallelQueue::new`]
+    pub unsafe fn begin(&mut self, device: &Device) {
+        if self.active.is_some() {
+            return;
+        }
+        let time = NonZeroU64::new_unchecked(self.shared.next_time.fetch_add(1, Ordering::Relaxed));
+        let cmd = match self.spare_cmds.pop() {
+            Some(cmd) => cmd,
+            None => {
+                let complete = device
+                    .get_semaphore_counter_value(self.shared.semaphore)
+                    .unwrap();
+                while self
+                    .in_flight
+                    .front()
+                    .map_or(false, |work| work.time.get() <= complete)
+                {
+                    self.spare_cmds
+                        .push(self.in_flight.pop_front().unwrap().cmd);
+                }
+                if self.spare_cmds.is_empty() {
+                    self.spare_cmds.extend(
+                        device
+                            .allocate_command_buffers(
+                                &vk::CommandBufferAllocateInfo::builder()
+                                    .command_pool(self.cmd_pool)
+                                    .command_buffer_count(32),
```

Comment:
How cheap are command buffers? Would it make sense to make `command_buffer_count` configurable? 
(same with `HandleSeed` above)

Reply:
Should be pretty cheap. This can be revisited in the unlikely event it becomes an issue. I try to avoid configuration knobs where there's no particular reason to expect a user might know a particularly appropriate value for their case.

Reply:
That's fair, thanks!